main/read.c

```diff
@@ -669,68 +658,12 @@ static void fileNewline (void)
 	if (BackupFile.mio == NULL)
 		appendLineFposMap (&File.lineFposMap, File.filePosition);
 
-	File.newLine = false;
 	File.input.lineNumber++;
 	File.source.lineNumber++;
 	DebugStatement ( if (Option.breakLine == File.input.lineNumber) lineBreak (); )
 	DebugStatement ( debugPrintf (DEBUG_RAW, "%6ld: ", File.input.lineNumber); )
 }
 
-/*  This function reads a single character from the stream, performing newline
- *  canonicalization.
- */
-static int iFileGetc (void)
-{
-	int	c;
-readnext:
-	c = mio_getc (File.mio);
-
-	/*	If previous character was a newline, then we're starting a line.
-	 */
-	if (File.newLine  &&  c != EOF)
-	{
-		fileNewline ();
-		if (c == '#'  &&  Option.lineDirectives)
-		{
-			if (parseLineDirective ())
-				goto readnext;
-			else
-			{
-				mio_setpos (File.mio, &StartOfLine);
-				c = mio_getc (File.mio);
-			}
-		}
-	}
-
-	if (c == EOF)
-		File.eof = true;
-	else if (c == NEWLINE)
-	{
-		File.newLine = true;
-		mio_getpos (File.mio, &StartOfLine);
-	}
-	else if (c == CRETURN)
-	{
-		/* Turn line breaks into a canonical form. The three commonly
-		 * used forms if line breaks: LF (UNIX/Mac OS X), CR (Mac OS 9),
-		 * and CR-LF (MS-DOS) are converted into a generic newline.
-		 */
```

Comment:
Done. I just modified the comment a bit because the part about normalizing CR (Mac OS 9) newlines wasn't quite correct (it was done only when macintosh macro was defined which I guess was defined under Mac OS 9 but the normalization didn't happen on other platforms).